Engie examples

moptis opened the pull request:
First pass at reorganization of Examples folder. Lots of cleaning and documentation to come, but this should give you an idea on where we're at. A brief summary:

1. Example notebooks labeled 00 through 05 starting with toolkit examples and then working through QC, AEP, TIE, Elec losses and finally gap analysis using Engie data.

2. I moved all old stuff to an 'old' folder which ultimately we'll delete when we're comfortable.

3. Eric - I made some changes to the project import script including the naming of the wind speed columns. I know it's not quite -25 compliant, but all our projects and code currently use this label. We can scope a PR to copy/replace this at a later time.

moptis commented:
Ahh shoot, moving things to the old folder messed up the testing I think.

jordanperr commented:
Yes - the tests all unzip the data which was in the examples folder. Also, the docs are automatically generated from the example notebooks. Messing with these paths will mess up the tests.

Here is the line which is causing the tests to fail: https://github.com/NREL/OpenOA/blob/75248b79354734e23184ee197e2221bea208f69b/.github/workflows/python-ci.yml#L29

Here is the line which is causing the docs to fail: https://github.com/NREL/OpenOA/blob/75248b79354734e23184ee197e2221bea208f69b/sphinx/conf.py#L25

Would you like me to try and fix these issues now by pointing the tests and the documentation to the "old" folder? Maybe it's better to leave the tests failing for now, and then fix them one-by-one as we convert them to use the Engie data.

jordanperr commented:
Also, the README should be updated to reflect any new unzipping / test running procedures.

jordanperr commented:
Please note - the github UI is incorrectly showing 15 changed files, when this is an artifact of a 3-way merge I did in the squashed commit.

nbodini commented on examples/03_turbine_ideal_energy.ipynb:
```diff
@@ -0,0 +1,405 @@
+    "       wind_bin_thresh = 2, \n",
+    "       correction_threshold = 0.90, \n",
+    "       enable_plotting = False,\n",
+    "       plot_dir = None)"
```
PR #77 now allows filtering parameters to be passed as arguments in the UQ = False.
Should I also move all the arguments back to run instead of init to make the new version backward compatible?

nbodini replied:
PR #77 should fix this.

moptis replied:
I can get 03_notebook to work by modifying the inputs in the init and run functions. Is PR 77 still needed?

ejsimley replied:
PR 77 does fix some other things too (like hardcoded parameters that are changed to arguments), so let's keep it.

nbodini commented on examples/05_eya_gap_analysis.ipynb:
```diff
@@ -0,0 +1,286 @@
+    "       wind_bin_thresh = 2, \n",
+    "       correction_threshold = 0.90, \n",
+    "       enable_plotting = False,\n",
+    "       plot_dir = None)"
```
PR #77 should fix this.

moptis replied:
I fixed the init and run methods in the notebook and it works now.

nbodini commented on examples/03_turbine_ideal_energy.ipynb:
```diff
@@ -0,0 +1,405 @@
+    },
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZIAAAFNCAYAAAAq6ukAAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjMsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+AADFEAAAgAElEQVR4nOy9eXhcZ3mwfz8zo5EtyZJly7LlVd4d2yHLccJA4AcJSxNoCCUsAUPSFkihoYXShaX9UpoPKLQfS4FAm7KFoARCEiBQQslCSAKZJB5ndeLEq7zJ1mprsaTZ3t8fzxlrJEuj0TLa/NzXNdfMvOe857wzGp3nPLs45zAMwzCM0RKY7AUYhmEY0xs ...
```
PR #77 should fix this too.

moptis replied:
Is this the daily plotting function? so hard to tell when commenting on a notebook. But yes this is still failing but I don't want to modify any code in the TIE method or we'll get a conflict down the road when we merge branches.

moptis commented on examples/project_ENGIE.py:
```diff
@@ -56,40 +56,45 @@ def __init__(self, path="data/la_haute_borne", name="Engie",
+        time = [s[11:19] for s in self._scada.df['Date_time']]
+        datetime = [date[s] + ' ' + time[s] for s in np.arange(len(date))]
+        self._scada.df['time'] = pd.to_datetime(datetime, format = "%Y-%m-%d %H:%M:%S")
+
```
@ejsimley The problem with your code is that the object that was produced didn't have the same date averaging capability. Like saying df.resample('H') wouldn't work on the scada data frame. So I just went with what I know. If we can adjust your code to give that time sampling functionality that would be ideal.

moptis replied:
Weird...your code works fine now. Not sure what I did earlier. Switching back to your earlier code!

ejsimley replied:
ok, weird. yeah I just tested df.resample() and it worked ok too.

moptis commented on examples/00_toolkit_examples.ipynb:
```diff
@@ -0,0 +1,409 @@
+   "outputs": [],
+   "source": [
+    "windspeed_filt2 = windspeed_filt1[~out_of_window]\n",
+    "power_kw_filt2 = power_kw_filt1[~out_of_window]"
```
ahh good catch!

moptis commented on examples/01_qc_data.ipynb:
```diff
@@ -0,0 +1,729 @@
+    }
+   ],
+   "source": [
+    "qc.daylight_savings_plot()"
```
Labels added to plots with next update.

moptis commented on examples/03_turbine_ideal_energy.ipynb:
```diff
@@ -0,0 +1,405 @@
+    }
+   ],
+   "source": [
+    "ta.plot_filtered_power_curves(save_folder = \"/Users/moptis/Documents/temp/\", output_to_terminal = True)"
```
frozen sensor flag, as to your earlier point regarding 2 decimal places.

ejsimley replied:
Ah, good point.

moptis commented on examples/04_electrical_losses.ipynb:
```diff
@@ -0,0 +1,201 @@
+    "monthly_merge = el._merge_df.resample('MS').sum()\n",
+    "plt.plot((monthly_merge['corrected_energy'] - monthly_merge['energy_kwh']) / monthly_merge['corrected_energy'] * 100)\n",
+    "plt.xlabel('Month')\n",
+    "plt.ylabel('Electrical Losses (%)')"
```
An outlier from your AEP sampling method, @ejsimley? Does it look suspicious?

moptis commented:
@ejsimley I've addressed all of your comments, save for the one error getting the plot_daily_fitting_results() function to work. I don't want to edit the turbine ideal energy method. But we can flag this later to check once this PR and the TIE PR are merged.

ejsimley commented:
@moptis, @nbodini I suggest we wait to merge in Nicola's augmented AEP code (should be in the next few days), then finalize the example notebooks. How about we change the TIE and elec loss notebooks to include uncertainty, and then for plant AEP analysis create one example that uses the original method (monthly, linear regression) and another example that highlights daily resolution, the additional regressors, and ML. Or any combination of inputs that you think makes sense as an example Nicola. Does this sound like a good plan?

I think we can take a couple weeks to finish the examples if needed.

moptis commented:
@ejsimley @jordanperr  

I beefed up the prose in the examples notebooks but am still just using monthly linear regression for AEP and not exploring uncertainty in TIE and electrical losses. This should be sufficient for V2, no?

ejsimley commented:
Thanks for updating the prose @moptis! 

@moptis and @nbodini Since you and Nicola put so much effort into daily/augmented AEP and uncertainty in TIE and elec losses, I think we should highlight them in the examples. I don't expect this to take too much time, especially since Nicola already wrote tests for these new classes. For example, showing the option to include uncertainty in electrical losses would probably just be one or two additional cells in the existing notebook.

In my opinion, since we can wait a couple weeks before "releasing" v2, we might as well showcase these new methods. Of course we can help you out with this. Thoughts?

moptis commented:
Looks good to me. I approve!
